trinity/utils/os.py

```diff
@@ -7,3 +9,15 @@ def get_open_fd_limit() -> int:
     """
     soft_limit, hard_limit = resource.getrlimit(resource.RLIMIT_NOFILE)
     return soft_limit
+
+
+def friendly_filename_or_url(value: str) -> str:
+    """
+    Normalize any string to be file name and URL friendly.
+    Convert to lowercase, remove non-alpha characters,
+    and convert spaces to hyphens.
+    """
+    value = str(unicodedata.normalize('NFKD', value).encode('ascii', 'ignore'))
```

Comment:
This is from https://stackoverflow.com/questions/295135/turn-a-string-into-a-valid-filename/295466#295466

Reply:
Can you embed that link into the comment so that this is documented in the code.